nilearn/input_data/tests/test_nifti_masker.py

```diff
@@ -315,3 +313,34 @@ def test_filter_and_mask():
 
     assert_raises_regex(DimensionError, "Data must be a 3D", filter_and_mask,
                          data_img, mask_img, params)
+
+
+def test_multi_nifti_masker_valid_cache():
+    mask = np.ones((10, 10, 1, 1))
+    niimg = np.ones((10, 10, 1, 1))
+    affine = np.eye(4)
+
+    mask_img = Nifti1Image(mask, affine)
+    niimg = Nifti1Image(niimg, affine)
+
+    temp_dir = mkdtemp()
+    with write_tmp_imgs(mask_img, niimg) as (mask_img_f, niimg_f):
+        mem = Memory(cachedir=temp_dir)
+        mem.clear()
+        for this_mask_img, this_niimg in [(mask_img_f, niimg_f),
+                                          (mask_img, niimg)
+                                          ]:
+            masker = NiftiMasker(memory=mem, memory_level=1, verbose=10,
+                                      mask_img=this_mask_img,
+                                      standardize=False)
+            masker.fit()
+            masker.transform(this_niimg)
+            masker.transform(this_niimg)
+            masker.fit()
+            masker.transform(this_niimg)
+            l = os.listdir(join(temp_dir, 'joblib', 'nilearn',
+                                'input_data',
+                                'nifti_masker', 'filter_and_mask'))
+            # Ensure that filter and mask has not dumped twice
+            # the same data
+    assert_equal(len(l), 3)
```

Comment:
I wanted to clear memory with a `mem.clear()` inside the for loop, but for some reason `func_code.py` does not appear in the second iteration if I do that -- which is bewildering.